biz.aQute.bndlib.tests/src/test/ProcessorTest.java

```diff
@@ -188,6 +191,44 @@ private void assertNative(String in, String filter, String... fixup) throws Exce
 		}
 	}
 
+	static class SeethroughProcessor extends Processor {
```

Comment:
Not sure what the purpose of this subclass is. It does not change the behavior of Processor at all. I think you can remove this subclass.

Reply:
The accessibility on `Processor.beginHandleErrors()`/`endHandleErrors()` is protected. We can't invoke these methods from within the test method on a bare `Processor` object. Subclassing it was the easiest way to gain direct access to the methods for testing, as subclasses are allowed to widen the accessibility of overridden methods (hence calling it "Seethrough" - you can see the methods that ordinarily would be).

To be fair, this is not immediately obvious. If it helps, I can add a comment to this effect to explain to the next developer why this was done.

Another alternative would be to use reflection to bypass the accessibility check, but unless you have a good reflection helper API on the test classpath this code wouldn't be much easier to read.

Reply:
Of course. I did not see that. I guess a comment would help to explain. Thanks for educating me!

Reply:
Actually, perhaps a better alternative would be to move the unit test into the `aQute.bnd.osgi` package, and then the protected methods would be accessible from the test method without me needing to override it. I'm happy to implement whatever you suggest.

Reply:
Use the subclass with a comment to explain why.